### PR TITLE
HTML5 Product Microdata display

### PR DESCRIPTION
**Pull request by toomanyenergydrinks**
Proposed fix for bug mentioned here https://github.com/spree/spree/issues/6839

This is my first pull req to a major repo, so hopefully I'm doing this right!

### CODE REVIEW

File: frontend/app/views/spree/shared/_products.html.erb [hunk 1]
```diff
@@ -24,8 +24,8 @@
         <div class="panel panel-default">
           <% cache(@taxon.present? ? [I18n.locale, current_currency, @taxon, product] : cache_key_for_product(product)) do %>
             <div class="panel-body text-center product-body">
-              <%= link_to small_image(product, itemprop: "image"), url, itemprop: 'url' %><br/>
-              <%= link_to truncate(product.name, length: 50), url, class: 'info', itemprop: "name", title: product.name %>
+              <%= link_to small_image(product, itemprop: "image") + content_tag(:span, truncate(product.name, length: 50), class: 'info', itemprop: "name", title: product.name), url, itemprop: 'url' %>
```
Comment (priyank-gupta) [inline frontend/app/views/spree/shared/_products.html.erb]: @rehan-work This seems a bit complicated. Instead we can use a block with link_to and add image and span there.

**Comment by priyank-gupta** [comment]
@rehan-work Not sure if we want to include microdata gem here.
@damianlegawiec @Mafi88 ?

**Comment by rehan-work** [comment]
Noted, will update the PR with the changes. I also realise adding a new gem is a significant change, I can continue to look into alternative ways to test the microdata if needed.

**Comment by Mafi88** [comment]
@rehan-work I think it would be nice, especially that the gem hasn't been updated for 3 years now.

**Comment by rehan-work** [comment]
Good point, thanks for the feedback! I'll work on removing the gem dependency from the tests.

**Comment by Mafi88** [comment]
@rehan-work Thanks!

**Comment by Mafi88** [comment]
@rehan-work Seems good to me, could you please address the houndci issues? And rebase & squash after this? Thanks!

**Comment by rehan-work** [comment]
I'm working on fixing those now, does the initial commit need test coverage for the testing the support functions?

**Comment by Mafi88** [comment]
I don't think so... @priyank-gupta, @damianlegawiec?

**Comment by priyank-gupta** [comment]
:+1:

**Pull request merged**